src/librustc_mir/build/expr/into.rs

```diff
@@ -120,15 +119,15 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                 this.cfg.terminate(block, source_info, TerminatorKind::SwitchInt {
                     discr: lhs,
                     switch_ty: this.hir.bool_ty(),
-                    values: vec![ConstVal::Bool(true)],
+                    values: BOOL_SWITCH_TRUE.clone(),
```

Comment:
Nit: I think I'd expect, when using switch, that we set it up as `false => ..., _ => ...`, since in C one normally interprets "non-zero == true". Maybe it leads to marginally more efficiency assembly.